lib/3scale/backend/async_redis/endpoint_helpers.rb

```diff
@@ -14,20 +15,40 @@ class << self
 
           # @param host [String]
           # @param port [Integer]
+          # @param path [String]
+          # @param ssl [Boolean]
           # @param ssl_params [Hash]
           # @return [Async::IO::Endpoint::Generic]
-          def prepare_endpoint(host, port, ssl = false, ssl_params = nil)
+          def prepare_endpoint(**kwargs)
+            return prepare_tcp_endpoint(**kwargs) if host_present?(kwargs[:host])
+
+            prepare_unix_endpoint(**kwargs)
```

Comment:
IMO would read nicer like:
```suggestion
            host_present?(kwargs[:host]) ? prepare_tcp_endpoint(**kwargs) : prepare_unix_endpoint(**kwargs)
```